### PR TITLE
Check multiple licenses for allowed classifications 

### PR DESCRIPTION
**Pull request by jonabc**
Fixes https://github.com/github/licensed/issues/186

This PR updates `licensed status` to not raise an error in the case that 
1. The `license` field is set as `other`
2. There are multiple `licenses` entries
3. All `licenses` entries match a license classification that are configured `allowed`

This is meant to address cases where a project has multiple licenses that are each allowed and specified separately.  It won't address cases where a single `LICENSE` file references multiple licenses.

Under the hood this uses licensee to re-parse the cached content using both license file parsing and readme file parsing logic, and picking the first non-`other` license found between the two.  This should be fully backwards compatible as it doesn't require any additional metadata to work. 
 An assumption is made that if the content is parseable by both `LicenseFile` and `ReadmeFile`, that they won't result in different license classifications.

### CODE REVIEW

**Comment by jonabc** [comment]
@mlinksva re: my last [comment](https://github.com/github/licensed/issues/186#issuecomment-534324174), I sidestepped the "which license classification do I edit" question by not adding any new fields, and evaluating the cached license text using licensee.

File: lib/licensed/commands/status.rb [hunk 1]
```diff
@@ -35,20 +35,52 @@ def evaluate_dependency(app, source, dependency, report)
...
+        # review is not needed if the top level license is allowed
+        return false if app.allowed?(cached_record["license"])
+
+        # the remaining checks are meant to allow records marked as "other"
+        # that have multiple licenses, all of which are allowed
+
+        # review is needed for non-"other" licenses
+        return true unless cached_record["license"] == "other"
+
+        licenses = cached_record.licenses.map { |license| license_from_text(license.text) }
```
Comment (mlinksva) [inline lib/licensed/commands/status.rb]: Ignorant question: can a license detected from a package manager file be in this list? Asking because it wouldn't be detected by `license_from_text` below.

**Comment by mlinksva** [comment]
That sounds like a good strategy. I made one likely irrelevant comment.

File: lib/licensed/commands/status.rb (hunk 1 above)
Comment (jonabc) [reply lib/licensed/commands/status.rb]: 👍 it won't be detected because package manager text [isn't cached](https://github.com/github/licensed/blob/4abf047b99910f5a28a93f599381ab5bd7cdf8dc/lib/licensed/dependency.rb#L75) under the `licenses` field.

That in itself is somewhat confusing as it's a one-time / hidden check.  That's been around since before I started maintaining licensed, but I'll log an issue to re-examine if we should track that in the cached data and the best way to represent that.

Comment (jonabc) [reply lib/licensed/commands/status.rb]: https://github.com/github/licensed/issues/189

**Comment by PabloLION** [comment]
Hi, @jonabc. I know this is a old PR and released in 2.5.0, but I still want to ask if there is some example? 
I found that you wrote this https://github.com/github/licensed/issues/186#issuecomment-533236835 but I'm not sure it is the correct final version. Could you confirm it please?
Best wishes

**Comment by jonabc** [comment]
@PabloLION :wave: the YAML syntax in the comment you linked isn't what ended up being used or implemented.  Individual files won't show their detected licenses.  This PR only changed the detection logic for when an "other" classification is determined as something the user needs to take some action on.

As an example, consider a project that references both BSD and MIT licenses, and both of those licenses are configured as `allowed` in the licensed configuration file.  Due to using two licenses, licensee will give the project a license classification of `other`.  Prior to this change that would have been called out as an error from `licensed status` and would have required the user to add the project to the `reviewed` licensed configuration.  With this change no error will be raised from `licensed status` and the user will not need to take any action.

This is referenced on the [status command documentation](https://github.com/github/licensed/blob/master/docs/commands/status.md), as the sub-bullet on case (4) for when the status command will raise an error.

**Comment by PabloLION** [comment]
It seems I am asking in the wrong place. 
Would you mind telling me how to add two license so that they can be detected correctly? 

## tl;dr
These issues has the same problem and I'm referring them so people from SEO can see it.
https://github.com/gogs/gogs/issues/2173
https://github.com/github/choosealicense.com/issues/606

**Comment by mlinksva** [comment]
@PabloLION this repo is **not** about the rule of how GitHub detects License of repos. https://github.com/licensee/licensee is the code for that. If there are multiple licenses present, GitHub won't detect any of them at the moment.

**Comment by PabloLION** [comment]
Yes. I see it. Thank you. I agree to close the discussion here.
